core/src/main/java/com/linecorp/armeria/server/throttling/FileBasedDynamicThrottlingStrategy.java

```diff
@@ -0,0 +1,150 @@
+                while (true) {
+                    try {
+                        WatchKey key = watchService.take();
+                        for (WatchEvent<?> event : key.pollEvents()) {
+                            // We only register "ENTRY_MODIFY" so the context is always a Path.
+                            final Path changed = (Path) event.context();
+                            if (changed.equals(settingFile.getFileName())) {
+                                updateThrottlingSettings();
+                                break;
+                            }
+                        }
+                        if (!key.reset()) {
+                            break;
+                        }
+                    } catch (InterruptedException e) {
+                        throw new IllegalStateException("Interrupted while taking from file.", e);
+                    }
+
+                }
+            } catch (ClosedWatchServiceException ignored) {
+                logger.info("Stopping hosts file watch thread.");
+            }
+        });
+        thread.setDaemon(true);
+        thread.start();
+        return thread;
+    }
+
+    private void updateThrottlingSettings() {
+        Map<String, String> newStrategies;
+        try {
+            newStrategies = OBJECT_MAPPER.readValue(settingFile.toFile(),
+                                                    new TypeReference<Map<String, String>>() {});
+        } catch (IOException e) {
+            logger.warn("Could not read hosts file", e);
+            return;
+        }
+
+        strategies = newStrategies.entrySet().stream()
+                                  .collect(toImmutableMap(Entry::getKey,
+                                                          e -> convertToThrottlingStrategy(e.getValue())));
+    }
+
+    private static ThrottlingStrategy convertToThrottlingStrategy(String throttlingSpec) {
+        List<String> spec = COLON_SPLITTER.splitToList(throttlingSpec);
+        if (spec.size() != 2) {
+            logger.warn("Invalid throttlingSpec: {}", throttlingSpec);
+            return ThrottlingStrategy.never();
+        }
+        switch (Ascii.toLowerCase(spec.get(0))) {
+            case "always":
+                return ThrottlingStrategy.always();
+            case "never":
+                return ThrottlingStrategy.never();
+            case "sample":
+                float rate = Floats.tryParse(spec.get(1));
+                return (ctx, req) -> completedFuture(rate > ThreadLocalRandom.current().nextFloat());
+            default:
+                throw new IllegalArgumentException("Unknown key " + spec.get(0));
+        }
+    }
+
+    @Override
+    public CompletableFuture<Boolean> shouldThrottle(ServiceRequestContext ctx, Request req) {
+        String method = req instanceof RpcRequest ? ((RpcRequest) req).method() : ctx.method().name();
+        return strategies.getOrDefault(method, ThrottlingStrategy.never()).shouldThrottle(ctx, req);
```

Comment:
Just curious. Is this enough to throttle a request based on the request method? What do you think about adding a simple path pattern as a key?